src/software/gui/standalone_simulator/widgets/standalone_simulator_draw_function_visualizer.h

```diff
@@ -45,14 +47,26 @@ class StandaloneSimulatorDrawFunctionVisualizer : public DrawFunctionVisualizer
     // The robot currently being placed by the user, if any
     std::weak_ptr<PhysicsRobot> robot;
 
+    // Boolean to check if user shift clicked
+    bool shift_clicked = false;
+
     // Boolean to check if the user ctrl clicked
     bool ctrl_clicked = false;
 
+    // Boolean to check if user has r pressed
+    bool r_clicked = false;
+
     // The initial point when adding force to the ball, if any
     Point initial_point;
 
     // The final point when adding force to the ball, if any
     Point final_point;
 
+    // The initial point when orienting robot
+    Point rob_initial_point;
+
+    // The final point with orienting robot
+    Point rob_final_point;
+
```

Comment:
why can't we use initial_point/final_point here?

Reply:
The reason I used separate fields is that using initial and final points creates a line on the screen, as if the robot was the ball. 
![Screenshot from 2021-03-12 15-55-04](https://user-images.githubusercontent.com/48522062/111010421-55c9b380-834b-11eb-9263-4cd396cd3285.png)

Reply:
yikes we need to fix that

Reply:
I think the problem might be a little worse than I thought, I tried to revert the change and now it just draws a line on the screen, doesn't even rotate

Reply:
`getDrawBallVelocityFunction` can be detached from `initial_point` and `final_point`if we store the current ball velocity draw function as a private member and only initialize it if we're control clicking a ball. Then, `getDrawBallVelocityFunction` can just return this private member.